addons/calendar/calendar.py

```diff
@@ -884,6 +884,13 @@ def _set_date(self, cr, uid, values, id=False, context=None):
                 duration = float(diff.days) * 24 + (float(diff.seconds) / 3600)
                 values['duration'] = round(duration, 2)
 
+    def _is_highlighted(self, cr, uid, ids, field_name, arg, context=None):
+        res = dict.fromkeys(ids, False)
+        for event in self.browse(cr, uid, ids, context=context):
+            if event.partner_ids.filtered(lambda s: s.id == context.get('active_id')) and context.get('active_model') == 'res.partner':
```

Comment:
I would keep this model checking condition above the loop, this way we will save some iterations when it is called with crm.lead model